crypto/rsa/rsa_lib.c

```diff
@@ -1362,9 +1364,15 @@ int EVP_PKEY_CTX_set_rsa_keygen_pubexp(EVP_PKEY_CTX *ctx, BIGNUM *pubexp)
         return -1;
 
     /* TODO(3.0): Remove this eventually when no more legacy */
-    if (ctx->op.keymgmt.genctx == NULL)
-        return EVP_PKEY_CTX_ctrl(ctx, EVP_PKEY_RSA, EVP_PKEY_OP_KEYGEN,
-                                 EVP_PKEY_CTRL_RSA_KEYGEN_PUBEXP, 0, pubexp);
+    if (ctx->op.keymgmt.genctx == NULL) {
+        if (copy == 1)
+            pubexp = BN_dup(pubexp);
+        ret = EVP_PKEY_CTX_ctrl(ctx, EVP_PKEY_RSA, EVP_PKEY_OP_KEYGEN,
```

Comment:
Should it be doing the dup in here or in the ctrl?

```
ret = EVP_PKEY_CTX_ctrl(ctx, EVP_PKEY_RSA, EVP_PKEY_OP_KEYGEN,
                                          EVP_PKEY_CTRL_RSA_KEYGEN_PUBEXP,
                                          copy, pubexp);
```

Reply:
EVP_PKEY_CTX_ctrl() is technically a public API, right? So changing behavior there to do a copy instead of "taking custody" as it currently does would kind of be introducing into another API the same exact issue this change is trying to solve: preexisting callers who expect that they are no longer responsible for freeing the BIGNUM once it has been "passed off" to the API would suddenly start leaking memory.